src/test/java/co/rsk/federate/FedNodeRunnerTest.java

```diff
@@ -0,0 +1,219 @@
+import com.typesafe.config.Config;
+import org.ethereum.config.Constants;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by Kelvin Isievwore on 24/04/2023.
+ */
+public class FedNodeRunnerTest {
+    private FedNodeRunner fedNodeRunner;
+    private FedNodeSystemProperties fedNodeSystemProperties;
+    private Config config;
+
+    @Before
+    public void setUp() {
+        BridgeConstants bridgeConstants = mock(BridgeConstants.class);
+        Constants constants = mock(Constants.class);
+        fedNodeSystemProperties = mock(FedNodeSystemProperties.class);
+        when(fedNodeSystemProperties.getNetworkConstants()).thenReturn(constants);
+        when(constants.getBridgeConstants()).thenReturn(bridgeConstants);
+        when(bridgeConstants.getBtcParamsString()).thenReturn(NetworkParameters.ID_REGTEST);
+
+        config = mock(Config.class);
+        when(config.getString("host")).thenReturn("127.0.0.1");
+        when(config.getInt("port")).thenReturn(9999);
+        when(config.getString("keyId")).thenReturn("rsk");
+        when(config.getString("path")).thenReturn("reg1.key");
+
+        fedNodeRunner = new FedNodeRunner(
+            mock(BtcToRskClient.class),
+            mock(BtcToRskClient.class),
+            mock(BtcReleaseClient.class),
+            mock(FederationWatcher.class),
+            mock(FederatorSupport.class),
+            mock(FederateLogger.class),
+            mock(RskLogMonitor.class),
+            mock(NodeRunner.class),
+            fedNodeSystemProperties,
+            mock(FedNodeContext.class)
+        );
+    }
+
+    @Test
+    public void test_with_KeyFile_Config_Ok() throws Exception {
+        SignerConfig btcSignerConfig = getBTCSignerConfig();
+        SignerConfig rskSignerConfig = getRSKSignerConfig();
+        SignerConfig mstSignerConfig = getMSTSignerConfig();
+        when(fedNodeSystemProperties.signerConfig(FedNodeRunner.BTC_KEY_ID.getId())).thenReturn(btcSignerConfig);
+        when(fedNodeSystemProperties.signerConfig(FedNodeRunner.RSK_KEY_ID.getId())).thenReturn(rskSignerConfig);
+        when(fedNodeSystemProperties.signerConfig(FedNodeRunner.MST_KEY_ID.getId())).thenReturn(mstSignerConfig);
+
+        fedNodeRunner.run();
+
+        ECDSASigner signer = TestUtils.getInternalState(fedNodeRunner, "signer");
+        assertNotNull(signer);
+        assertTrue(signer.canSignWith(FedNodeRunner.BTC_KEY_ID));
+        assertTrue(signer.canSignWith(FedNodeRunner.RSK_KEY_ID));
+        assertTrue(signer.canSignWith(FedNodeRunner.MST_KEY_ID));
+
+        assertTrue(signer instanceof ECDSACompositeSigner);
+        ECDSACompositeSigner compositeSigner = (ECDSACompositeSigner) signer;
+        List<ECDSASigner> signers = TestUtils.getInternalState(compositeSigner, "signers");
+        assertEquals(3, signers.size());
```

Comment:
We could also assert the signers are key file signers. instance of ECDSASignerFromFileKey

And getVersion should return 1